homeassistant/components/mqtt/__init__.py

```diff
@@ -215,8 +227,14 @@ def setup(hass, config):
 
     global MQTT_CLIENT
     try:
-        MQTT_CLIENT = MQTT(hass, broker, port, client_id, keepalive, username,
-                           password, certificate, protocol)
+        # Embedded broker doesn't have some ssl variables
+        if broker_in_conf:
```

Comment:
This shouldn't be necessary. Just make sure you set the three variables to `None` if the embedded broker is being used.